processor-api/src/main/java/com/synopsys/integration/alert/processor/api/distribute/ProviderMessageDistributor.java

```diff
@@ -44,31 +38,24 @@ public ProviderMessageDistributor(AuditAccessor auditAccessor, EventManager even
     }
 
     public void distribute(ProcessedNotificationDetails processedNotificationDetails, ProcessedProviderMessageHolder providerMessageHolder) {
-        UUID jobId = processedNotificationDetails.getJobId();
-        Long auditId = auditAccessor.findOrCreatePendingAuditEntryForJob(jobId, processedNotificationDetails.getNotificationIds());
-
-        // FIXME update event handling to accept the new models
-        List<ProjectMessage> projectMessages = extractProviderMessages(providerMessageHolder.getProcessedProjectMessages());
-        List<SimpleMessage> simpleMessages = extractProviderMessages(providerMessageHolder.getProcessedSimpleMessages());
-
-        ProviderMessageHolder tempProviderMessageHolder = new ProviderMessageHolder(projectMessages, simpleMessages);
-
         String channelName = processedNotificationDetails.getChannelName();
         ChannelKey destinationKey = ChannelKeys.getChannelKey(channelName);
-        if (null != destinationKey) {
-            DistributionEvent event = new DistributionEvent(destinationKey, jobId, auditId, tempProviderMessageHolder);
-            logger.info("Sending {}. Event ID: {}. Job ID: {}. Destination: {}", EVENT_CLASS_NAME, event.getEventId(), jobId, destinationKey);
-            eventManager.sendEvent(event);
-        } else {
+        if (null == destinationKey) {
             logger.warn("Unable to send {}. No {} with the name {} exists", EVENT_CLASS_NAME, DESTINATION_WRAPPER_CLASS_NAME, channelName);
+            return;
+        }
+
+        for (ProcessedProviderMessageHolder singleMessageHolder : providerMessageHolder.expand()) {
+            distributeIndividually(processedNotificationDetails.getJobId(), destinationKey, singleMessageHolder);
         }
     }
 
-    private <T extends ProviderMessage<T>> List<T> extractProviderMessages(List<ProcessedProviderMessage<T>> processedMessages) {
-        return processedMessages
-                   .stream()
-                   .map(ProcessedProviderMessage::getProviderMessage)
-                   .collect(Collectors.toList());
+    public void distributeIndividually(UUID jobId, ChannelKey destinationKey, ProcessedProviderMessageHolder providerMessageHolder) {
+        Set<Long> notificationIds = providerMessageHolder.extractAllNotificationIds();
+        Long auditId = auditAccessor.findOrCreatePendingAuditEntryForJob(jobId, notificationIds);
+        DistributionEvent event = new DistributionEvent(destinationKey, jobId, auditId, providerMessageHolder.toProviderMessageHolder());
```

Comment:
can we rename providerMessageHolder to processedMessageHolder?  It was confusing at first glance seeing providerMessageHolder.toProviderMessageHolder().  Made me ask the question isn't it a provider message holder already by the name.  It just gave me a pause looking at the code.  If you have a better variable name great.   If not it isn't worth holding up the check in of this code.